src/java/picard/sam/markduplicates/util/ReadEndsForMarkDuplicates.java

```diff
@@ -71,4 +71,9 @@ public ReadEndsForMarkDuplicates(final ReadEndsForMarkDuplicates read) {
         this.read1IndexInFile = read.read1IndexInFile;
         this.read2IndexInFile = read.read2IndexInFile;
     }
+
+    @Override
+    public ReadEndsForMarkDuplicates clone() {
```

Comment:
Your call, but consider adding `implements Cloneable` to each class that implements `clone`.